src/radiation/radiation.cpp

```diff
@@ -569,34 +560,48 @@ void ablate::radiation::Radiation::EvaluateGains(Vec solVec, ablate::domain::Fie
 
     // Get access to the absorption function
     auto absorptivityFunctionContext = absorptivityFunction.context.get();
+    auto emissivityFunctionContext = emissivityFunction.context.get();
 
     // Start by marching over all rays in this rank
-    for (std::size_t r = 0; r < raySegments.size(); r++) {
-        // zero our the calculation
-        auto& raySegmentsCalculation = raySegmentsCalculations[r];
-        raySegmentsCalculation.Ij = 0.0;
-        raySegmentsCalculation.Krad = 1.0;
+    for (std::size_t raySegmentIndex = 0; raySegmentIndex < raySegments.size(); ++raySegmentIndex) {
+        //! Zero this ray segment for all wavelengths
+        for (unsigned short int wavelengthIndex = 0; wavelengthIndex < absorptivityFunction.propertySize; wavelengthIndex++) {  //! Iterate through every wavelength entry in this ray segment
```

Comment:
What type is the propertySize?

Reply:
It is a PetscInt. Maybe this can be a smaller data type?

Reply:
You probably want to match absorptivityFunction.propertySize with unsigned short int.  I think that propertySize is a PetscInt.  PetscInt type changes depending upon how petsc is built.